game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/SupportCalculator.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+import org.triplea.java.collections.IntegerMap;
+
+/** Calculates how much support units can give */
+@Value
+@Getter(value = AccessLevel.PACKAGE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class SupportCalculator {
+
+  @Getter Map<UnitSupportAttachment.BonusType, List<UnitSupportAttachment>> supportRules;
+  Map<UnitSupportAttachment, IntegerMap<Unit>> supportUnits;
+  boolean defence;
+  boolean allies;
+
+  /**
+   * @param defence are the receiving units defending?
+   * @param allies are the receiving units allied to the giving units?
+   */
+  public SupportCalculator(
```

Comment:
Codacy found an issue: [The constructor 'SupportCalculator(Collection, Collection, boolean, boolean)' has an NPath complexity of 1203, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6326866)